include/Yash.h

```diff
@@ -311,78 +310,114 @@ class Yash {
             }
         }
 
-        printDescriptions();
+        printBasedOnInput(AutoCompletionType::NewLine);
         print(m_prompt.c_str());
     }
 
-    void printCommand()
+    void printInputCommand()
     {
         print(s_clearLine);
         print(m_prompt.c_str());
-        print(m_command.c_str());
+        print(m_inputCommand.c_str());
     }
 
-    void printCommands(const std::map<std::string, std::string>& descriptions)
+    void printNameAndDescription(const std::string_view name, const std::string_view desc, size_t allignmentSize)
     {
-        size_t maxCommandSize { std::accumulate(begin(descriptions), end(descriptions), 0u, [](size_t max, const auto& desc) {
-            return std::max(max, desc.first.size());
-        }) };
-
-        for (const auto& [command, description] : descriptions) {
-            std::string alignment((maxCommandSize + 2) - command.size(), ' ');
-            std::string line;
-            line.reserve(command.size() + alignment.size() + description.size() + 2);
-            line.append(command).append(alignment).append(description).append("\r\n");
-            print(line.c_str());
-        }
+        print(name.data());
+
+        for (size_t start = 0; start < ((allignmentSize + 2) - name.size()); start++)
+            print(" ");
```

Comment:
Won't this change result in way more IO (typically at least)? I assume the call to `print` will often actually send to some physical device like UART or dump it to a screen or something

Reply:
Good observation :+1: Yes, but we actually do this already on every single input. I can change it to a string but it will instead do an allocation; not sure what's the worst :)